### PR TITLE
JS code organization

### LINKED ISSUE #1152: Better structure of JS files
Currently a lot of the JS code lives in a single file: `storefront.js` (and `dashboard.js` for dashboard respectively). We should think of better organization of this code and split it in some separate modules. I would group the scripts in a similar manner to how styles are grouped. E.g. in storefront we should have files like`cart.js`, `checkout.js` etc. The same logic should be applied to dashboard scripts.

### PR DESCRIPTION
**Pull request by dominik-zeglen**
Closes #1152

### CODE REVIEW

File: saleor/static/js/components/mobile/filters-menu.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default $(document).ready((e) => {
+  $('.filters-menu').on('click', (e) => {
+    const t = $('.filters-menu__body');
```
Comment (elwoodxblues) [inline saleor/static/js/components/mobile/filters-menu.js]: Some more descriptive variable name would be nice.

File: saleor/static/js/components/product-filters.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+export default $(document).ready((e) => {
+  $('.toggle-filter').each(function () {
+    let icon = $(this).find('.collapse-filters-icon');
+    let ele = $(this).find('.filter-form-field');
```
Comment (elwoodxblues) [inline saleor/static/js/components/product-filters.js]: `ele` also isn't the best name. Names like `e` or `el` are okay in anonymous functions in obvious context, here the name should definitely refer to the DOM element it holds.

File: saleor/static/js/components/sorter.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export default $(document).ready((e) => {
+  $(document).ready((e) => {
+    $('.sort-by button').on('click', (e) => {
+      const t = $(e.currentTarget).parent();
+      const l = t.find('.sort-list');
```
Comment (elwoodxblues) [inline saleor/static/js/components/sorter.js]: The same in here.

**Comment by elwoodxblues** [comment]
@dominik-zeglen This PR looks very good, I think we could renaming some of the files and merging few others into one to make it clearer:
- `switch.js` name doesn't say much about what kind of switc is refers to, I would rather name it `bulk-actions.js`
- `sorter.js`, `product-filters.js`, `filters-menu.js` all relate to filtering on category page, maybe we could merge it into one file e.g. `category-page.js`?
- I think `misc.js` and `utils.js` in dashboard could be merged into one file

What do you think?

**Comment by dominik-zeglen** [comment]
@elwoodxblues Thanks for your feedback and I totally agree with renaming `switch.js` to `bulk-actions.js` - I really, really lack linguistic creativity :)

Maybe I am too literal, but sorting isn't the same thing as filtering and therefore shouldn't be merged together into one file - especially that `sorter.js` and `product-filters` describe two different widgets on the page. But I think it's good idea to merge `product-filters` with `filters-menu`, they relate to the same element.

`misc.js`and `utils.js` serve different purposes. `misc.js` provides "safe place" :) for actions that are one-liners or just hard to fit anywhere. On the other side, `utils.js` contains functions (and constants if there were any) that refer to more than widget and do not belong to any of them in obvious way.

**Comment by elwoodxblues** [comment]
I think files such as `utils.js` always end up as a place where you put any piece of code that doesn't match anywhere else (at least for me it always looked so). But I agree with your reasoning, maybe if we want to keep order in those files this is the way to go.

**Pull request merged**